source/initial_temperature/adiabatic.cc

```diff
@@ -33,17 +33,48 @@ namespace aspect
 {
   namespace InitialTemperature
   {
+
+    template <int dim>
+    Adiabatic<dim>::Adiabatic ()
+      :
+      surface_boundary_id(numbers::invalid_unsigned_int)
+    {}
+
+    template <int dim>
+    void
+    Adiabatic<dim>::initialize ()
+    {
+      // Find the boundary indicator that represents the surface
+      surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(surface_boundary_id);
+
+      // The input ascii table contains one data column (LAB depths(m)) in addition to the coordinate columns.
+      Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set,
+                                                    1);
+    }
+
     template <int dim>
     double
     Adiabatic<dim>::
     initial_temperature (const Point<dim> &position) const
     {
-      // convert input ages to seconds
-      const double age_top =    (this->convert_output_to_years() ? age_top_boundary_layer * year_in_seconds
-                                 : age_top_boundary_layer);
+      double age_top = 0;
       const double age_bottom = (this->convert_output_to_years() ? age_bottom_boundary_layer * year_in_seconds
                                  : age_bottom_boundary_layer);
-
+      if (read_from_ascii_file)
+        {
+          // The input ascii contains the age of the seafloor. User must provide ages in seconds
+          age_top = Utilities::AsciiDataBoundary<dim>::get_data_component(surface_boundary_id,
+                                                                          position,
+                                                                          0);
```

Comment:
I think it would be better to allow it to input the age in years. 
```suggestion
          const double age_in_seconds = Utilities::AsciiDataBoundary<dim>::get_data_component(surface_boundary_id,
                                                                          position,
                                                                          0);
          age_top = this->convert_output_to_years() ? age_in_seconds * year_in_seconds
                        : age_in_seconds);
```

Reply:
This still gives an error regarding the surface_boundary_set not being declared

```
use of undeclared identifier 'surface_boundary_set'; did you mean 'surface_boundary_id'?
          Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set, 1);
```
It's not recognizing it because it's defined in the `initialize` function instead of the `initial_temperature` I guess?

Reply:
@danieldouglas92 - was this comment in reference to the discussion above?

If so, can you confirm that the code gives the error listed above even after implementing?:
```
      if (read_from_ascii_file)
        Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set,
```